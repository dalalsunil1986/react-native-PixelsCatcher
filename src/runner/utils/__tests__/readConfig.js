/* @flow */
const fs = require('fs');

const readConfig = require('../readConfig');

jest.mock('fs', () => ({
  existsSync: jest.fn(),
  readFileSync: jest.fn(),
}));
jest.mock('path', () => ({
  join: jest.fn((...args: any) => args.join('/')),
}));

(process: any).exit = jest.fn();
(process: any).cwd = jest.fn(() => 'path_to_file');

describe('readConfig', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('read config from package.json', () => {
    const rawConfig = '{"PixelsCatcher":{"android":{"_content":"platform '
      + 'config content from package.json"}}}';

    (fs.existsSync: any).mockImplementationOnce(() => true);
    (fs.readFileSync: any).mockImplementationOnce(() => rawConfig);

    const config = readConfig();

    expect(config).toMatchSnapshot();
  });

  it('read config from pixels-catcher.json', () => {
    // package.json mock
    (fs.existsSync: any).mockImplementationOnce(() => true);
    (fs.readFileSync: any).mockImplementationOnce(() => '{}');
    // pixels-catcher.json mock
    const rawConfig = '{"ios":{"_content":"platform config content from '
      + 'pixels-catcher.json"}}';
    (fs.existsSync: any).mockImplementationOnce(() => true);
    (fs.readFileSync: any).mockImplementationOnce(() => rawConfig);

    const config = readConfig();

    expect(config).toMatchSnapshot();
  });
});
