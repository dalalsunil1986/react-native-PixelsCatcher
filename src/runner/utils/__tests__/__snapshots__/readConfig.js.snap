// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readConfig read config from package.json 1`] = `
Object {
  "android": Object {
    "_content": "platform config content from package.json",
  },
}
`;

exports[`readConfig read config from pixels-catcher.json 1`] = `
Object {
  "ios": Object {
    "_content": "platform config content from pixels-catcher.json",
  },
}
`;
